### ISSUE TITLE
Poor formatting rule for strings wrapped in parens when using testing dsl from kotest

### ISSUE THREAD

**Issue by fuelyourdev** [pinterest/ktlint#810]
The bug you're experiencing might have already be reported!
Please search in the [issues](https://github.com/pinterest/ktlint/issues) before creating one.

## Expected Behavior

In Kotest, using the StringSpec, you are given a dsl that allows you to write your test names as strings, like this:

```kotlin
class MyTests : StringSpec({
    "False should not equal true" {
        false shouldNotBe true
    }
})
```

And this is great, because this allows you to be much more descriptive and clear about your tests, but when your test name/description gets long enough to where you need to wrap it on to multiple lines, you then need to surround the string with parenthesis in order for the dsl to work (order of operations, function calls before string concatenation). With this being the case, I would imagine that formatting as follows would be most in line with the general formatting rules:

```kotlin
class MyTests : StringSpec({
    (
        "This is a test with a very long and descriptive name with " +
            "information describing a complicated test"
    ) {
        //...
        result shouldBe expected
    }
})
```

## Observed Behavior

But if I format the code as is shown above for long strings, then I get this error message from ktlint:

```
Lint error > [indent] Unexpected indentation (4) (should be 8)
```

ktlint expects the code to be formatted as so:

```kotlin
class MyTests : StringSpec({
    (
        "This is a test with a very long and descriptive name with " +
            "information describing a complicated test"
        ) {
        //...
        result shouldBe expected
    }
})
```

At which point in time the error message goes away. Ultimately it just seem counter to how parenthesis and curly braces work in other situations.

## Steps to Reproduce

Run ktlint against this code:

```kotlin
class MyTests : StringSpec({
    (
        "This is a test with a very long and descriptive name with " +
            "information describing a complicated test"
    ) {
        //...
        result shouldBe expected
    }
})
```

## Your Environment

* Version of ktlint used:
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
* Version of Gradle used (if applicable):
* Operating System and version:
* Link to your project (if it's a public repository):

Using kotlin 1.3.72
Using kotlinter 2.4.1 (gradle plugin), which uses ktlint 0.37.2
Using gradle 6.1
Using kotest-runner-junit5 4.0.5 and kotest-assertions-core-jvm 4.0.5
Running on Windows 10 in WSL2 Ubuntu

**Comment by romtsn** [comment]
Thanks for reporting. I have a feeling that this is the same as #796, yet I'm not sure and need to check

**Comment by romtsn** [comment]
Actually, I have just checked in detail and this turns out to be expected behavior - if you try to re-format it in IDEA, you'll get the same result, we even have special handling for this exact case:
https://github.com/pinterest/ktlint/blob/d23e97603ac4e296b36a9939ee59d53fbb0796da/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/IndentationRule.kt#L963-L974

It is unfortunate and has to be fixed on IDEA's side first, so might worth filing the issue on youtrack.

**Comment by romtsn** [comment]
Closing this until it's clear whether it's a bug or not. For now, it seems to be expected behavior. Will monitor the youtrack issue closely
